server/fishtest/api.py

```diff
@@ -347,6 +347,91 @@ def get_elo(self):
         run["elo"] = a
         return run
 
+    @view_config(route_name="api_calc_elo")
+    def calc_elo(self):
+        W = self.request.params.get("W")
+        D = self.request.params.get("D")
+        L = self.request.params.get("L")
+        LL = self.request.params.get("LL")
+        LD = self.request.params.get("LD")
+        DDWL = self.request.params.get("DDWL")
+        WD = self.request.params.get("WD")
+        WW = self.request.params.get("WW")
+        elo0 = self.request.params.get("elo0")
+        elo1 = self.request.params.get("elo1")
+        elo_model = self.request.params.get("elo_model", "normalized")
+
+        if elo_model not in ["BayesElo", "logistic", "normalized"]:
+            self.handle_error(
+                "Valid Elo models are: BayesElo, logistic, and normalized."
+            )
+
+        is_ptnml = all(
+            value is not None
+            and (
+                (
+                    isinstance(value, (int, float))
+                    or isinstance(value, str)
+                    and value.replace(".", "").replace("-", "").isdigit()
+                )
+                and float(value) >= 0
```

Comment:
If I'm not wrong `elo0` and/or `elo1` can be negative, same in `is_wdl`

Reply:
Oh that's an Oopsie!